parser/src/main/scala/play/twirl/parser/TwirlParser.scala

```diff
@@ -492,7 +492,8 @@ class TwirlParser(val shouldParseInclusiveDot: Boolean) {
   def blockArgs(): PosString = {
     val p = input.offset()
     val result = anyUntil("=>", inclusive = true)
-    if (result.endsWith("=>") && !result.contains("\n"))
+    if (result.endsWith("=>") && !result.contains("\n") && !result.contains("{") && !result.contains("}") &&
```

Comment:
This is now hard to reason about. Maybe extract groups of conditions as small methods?

Reply:
@marcospereira Thanks, see the commit I added.